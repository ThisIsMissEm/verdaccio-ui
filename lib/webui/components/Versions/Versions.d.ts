/**
 * @prettier
 */
import React from 'react';
declare class Versions extends React.PureComponent<any> {
    render(): JSX.Element;
    renderPackageList: (packages: any, isVersion?: boolean, timeMap?: Object) => JSX.Element;
    renderContent(packageMeta: any): JSX.Element;
}
export default Versions;
