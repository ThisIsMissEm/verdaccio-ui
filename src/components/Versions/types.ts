/**
 * @prettier
 */

export interface Versions {
  [name: string]: {
    author: string;
    dependencies: Array<{ [key: string]: string }>;
    description: string;
    keywords: Array<string>;
    license: string;
    main: string;
    name: string;
    version: string;
    scripts: Array<{ [key: string]: string }>;
    dist: Array<{ [key: string]: string }>;
  };
}

export interface DistTag {
  [key: string]: {
    [key: string]: string;
  };
}
