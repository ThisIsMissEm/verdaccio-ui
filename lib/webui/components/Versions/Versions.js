"use strict";
/**
 * @prettier
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Version_1 = require("../../pages/version/Version");
const package_1 = require("../../utils/package");
const styles_1 = require("./styles");
const List_1 = __importDefault(require("@material-ui/core/List"));
const ListItem_1 = __importDefault(require("@material-ui/core/ListItem"));
const react_1 = __importDefault(require("react"));
const constants_1 = require("../../../lib/constants");
const NOT_AVAILABLE = 'Not available';
class Versions extends react_1.default.PureComponent {
    constructor() {
        super(...arguments);
        this.renderPackageList = (packages, isVersion = false, timeMap = {}) => {
            return (react_1.default.createElement(List_1.default, null, Object.keys(packages)
                .reverse()
                .map(version => (react_1.default.createElement(ListItem_1.default, { className: "version-item", key: version },
                react_1.default.createElement(styles_1.ListItemText, null, version),
                react_1.default.createElement(styles_1.Spacer, null),
                isVersion && react_1.default.createElement(styles_1.ListItemText, null, timeMap[version] ? `${package_1.formatDateDistance(timeMap[version])} ago` : NOT_AVAILABLE),
                isVersion === false && react_1.default.createElement(styles_1.ListItemText, null, packages[version]))))));
        };
    }
    render() {
        return (react_1.default.createElement(Version_1.DetailContextConsumer, null, ({ packageMeta }) => {
            return this.renderContent(packageMeta);
        }));
    }
    renderContent(packageMeta) {
        const { versions = {}, time: timeMap = {}, [constants_1.DIST_TAGS]: distTags = {} } = packageMeta;
        return (react_1.default.createElement(react_1.default.Fragment, null,
            distTags && (react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(styles_1.Heading, { variant: "subheading" }, "Current Tags"),
                this.renderPackageList(distTags, false, timeMap))),
            versions && (react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(styles_1.Heading, { variant: "subheading" }, "Version History"),
                this.renderPackageList(versions, true, timeMap)))));
    }
}
exports.default = Versions;
