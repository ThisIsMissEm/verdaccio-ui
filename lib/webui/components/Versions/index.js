"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Versions_1 = require("./Versions");
exports.default = Versions_1.default;
