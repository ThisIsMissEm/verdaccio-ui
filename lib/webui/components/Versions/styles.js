"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_emotion_1 = __importDefault(require("react-emotion"));
const Typography_1 = __importDefault(require("@material-ui/core/Typography"));
const ListItemText_1 = __importDefault(require("@material-ui/core/ListItemText"));
exports.Heading = react_emotion_1.default(Typography_1.default) `
 && {
  font-weight: 700;
 }
`;
exports.Spacer = react_emotion_1.default('div') `
    flex: 1 1 auto;
    border-bottom: 1px dotted rgba(0, 0, 0, .2);
    white-space: nowrap;
    height: 0.5em;
`;
exports.ListItemText = react_emotion_1.default(ListItemText_1.default) `
    && {
      flex: none;
      color: black;
      opacity: .6;
    }
`;
