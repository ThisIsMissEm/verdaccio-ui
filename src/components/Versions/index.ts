/**
 * @prettier
 */

export { default } from './Versions';
