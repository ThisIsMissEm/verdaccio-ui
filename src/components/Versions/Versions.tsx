/**
 * @prettier
 */

import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Typography from '@material-ui/core/Typography';
import ListItemText from '@material-ui/core/ListItemText';

import React from 'react';

import { DetailContext, DetailContextProps } from '../../pages/version/Version';
import { formatDateDistance } from '../../utils/package';
import { Spacer } from './styles';
import { Version, DistTag } from './types';

const Versions: React.FC = () => {
  const renderPackageList = (versions: Version) => (
    <List>
      {Object.keys(versions)
        .reverse()
        .map(version => (
          <ListItem key={version}>
            <ListItemText>{version}</ListItemText>
            <Spacer />
            <ListItemText>{`${formatDateDistance('2017-10-26T09:03:15.044Z')} ago`}</ListItemText>
          </ListItem>
        ))}
    </List>
  );

  const renderDistTagsList = (distTags: DistTag) => (
    <List>
      {Object.keys(distTags)
        .reverse()
        .map(distTag => (
          <ListItem key={distTag}>
            <ListItemText>{distTag}</ListItemText>
            <Spacer />
            <ListItemText>{distTags[distTag]}</ListItemText>
          </ListItem>
        ))}
    </List>
  );

  const renderContent: React.FC<DetailContextProps> = ({ packageMeta: { versions, distTags } }) => (
    <>
      {distTags && (
        <>
          <Typography>{'Current Tags'}</Typography>
          {renderDistTagsList(distTags)}
        </>
      )}
      {versions && (
        <>
          <Typography>{'Version History'}</Typography>
          {renderPackageList(versions)}
        </>
      )}
    </>
  );

  return <DetailContext.Consumer>{context => context && renderContent(context)}</DetailContext.Consumer>;
};

export default Versions;
