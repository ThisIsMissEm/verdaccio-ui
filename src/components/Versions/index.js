/**
 * @prettier
 */

import { DetailContextConsumer } from '../../pages/version/Version';
import { formatDateDistance } from '../../utils/package';
import { Heading, Spacer, ListItemText } from './styles';
import List from '@material-ui/core/List/index';
import ListItem from '@material-ui/core/ListItem/index';
import React from 'react';
import { DIST_TAGS } from '../../../lib/constants';

class Versions extends React.PureComponent<any> {
  render() {
    return (
      <DetailContextConsumer>
        {({ packageMeta }) => {
          return this.renderContent(packageMeta[DIST_TAGS], packageMeta.versions);
        }}
      </DetailContextConsumer>
    );
  }

  renderPackageList = (packages: any, isVersion: boolean = false) => (
    <List>
      {Object.keys(packages)
        .reverse()
        .map(version => (
          <ListItem key={version}>
            <ListItemText>{version}</ListItemText>
            <Spacer />
            <ListItemText>{isVersion ? `${formatDateDistance('2017-10-26T09:03:15.044Z')} ago` : packages[version]}</ListItemText>
          </ListItem>
        ))}
    </List>
  );

  renderContent(distTags: object, versions: object) {
    return (
      <>
        {distTags && (
          <>
            <Heading variant={'subheading'}>{'Current Tags'}</Heading>
            {this.renderPackageList(distTags)}
          </>
        )}
        {versions && (
          <>
            <Heading variant={'subheading'}>{'Version History'}</Heading>
            {this.renderPackageList(versions, true)}
          </>
        )}
      </>
    );
  }
}

export default Versions;
