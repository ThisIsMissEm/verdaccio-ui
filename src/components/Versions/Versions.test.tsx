import React from 'react';
import { shallow } from 'enzyme';

import Versions from './Versions';

describe('<Versions /> component', () => {
  test('should render the component in default state', () => {
    const wrapper = shallow(<Versions />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
