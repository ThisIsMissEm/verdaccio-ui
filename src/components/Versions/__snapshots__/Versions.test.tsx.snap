// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Versions /> component should render the component in default state 1`] = `""`;
